### PR TITLE
rails5 updates

### PR DESCRIPTION
**Pull request by twalpole**
This updates refile to test with Rails master branch (rails 5) and fixes all failing tests.  There is still an error when running the tests on travis since for some reason it attempts to load a 'test' file after running rubocop -   I don't have time right now to figure out why, if anyone has any ideas please let me know.  Note currently this requires using a non-official branch of sinatra, hopefully that changes soon

### CODE REVIEW

**Comment by mshibuya** [comment]
:+1: need this too!

**Comment by atomn** [comment]
yep lookin forward to merge :+1:

File: gemfiles/Gemfile.rails-5.0-master [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+gem "rack", github: "rack/rack"
+gem "rspec-rails", github: "rspec/rspec-rails"
+gem "rspec", github: "rspec/rspec"
+gem "rspec-core", github: "rspec/rspec-core"
+gem "rspec-support", github: "rspec/rspec-support"
+gem "rspec-expectations", github: "rspec/rspec-expectations"
+gem "rspec-mocks", github: "rspec/rspec-mocks"
+# swap back to the sinatra repo once the sinatra send_file issue is fixed
+# gem "sinatra", github: "sinatra/sinatra", branch: "2.2.0-alpha"
+gem "sinatra", github: "kgrz/sinatra", branch: "use-rack-files-serving-method"
```
Comment (jackturnbull) [inline gemfiles/Gemfile.rails-5.0-master]: @twalpole; appears that https://github.com/sinatra/sinatra/pull/1045 has now been merged into [sinatra:2.2.0-alpha](https://github.com/sinatra/sinatra/tree/2.2.0-alpha).

Comment (twalpole) [inline gemfiles/Gemfile.rails-5.0-master]: Updated - thanks for notifying me

**Comment by jackturnbull** [comment]
Getting the following when using this branch with `rails-5.0.0.beta1`, likely due to the dependency in the Gemspec:
```
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rails (= 5.0.0.beta1) was resolved to 5.0.0.beta1, which depends on
      actionpack (= 5.0.0.beta1) was resolved to 5.0.0.beta1, which depends on
        rack (~> 2.x)

    refile was resolved to 0.6.2, which depends on
      sinatra (>= 1.4.5) was resolved to 1.4.5, which depends on
        rack (~> 1.4)

    devise was resolved to 3.5.3, which depends on
      warden (~> 1.2.3) was resolved to 1.2.4, which depends on
        rack (>= 1.0)

    devise was resolved to 3.5.3, which depends on
      warden (~> 1.2.3) was resolved to 1.2.4, which depends on
        rack (>= 1.0)
```

Hopefully not a case of waiting for Sinatra >= 2?

**Comment by twalpole** [comment]
@jackturnbull When using this branch you'll need to use the dependencies from  gemfiles/Gemfile.rails-5.0-master in your own Gemfile - The released version of Sinatra doesn't support rack 2

**Comment by jackturnbull** [comment]
@twalpole Of course! Knew I was missing something - thanks for the quick response again.

**Comment by jnicklas** [comment]
Thank you so much @twalpole, this is an awesome PR!

**Comment by darkcode85** [comment]
@twalpole @jackturnbull How I can use the dependencies from gemfiles/Gemfile.rails-5.0-master in my gemfile? I can not refile working using rails ```'>= 5.0.0.beta1', '< 5.1' ```

**Comment by twalpole** [comment]
@darkcode85 Look in  the gemfiles/Gemfile.rails-5.0-master file and compare the requirements to gems in your Gemfile.  Also look at the errors being output when you try and bundle, and see what packages are blocking higher required versions.  At the very least you probably need

    gem "sinatra", github: "sinatra/sinatra", branch: "2.2.0-alpha"

unless sinatra has realeased a version with full rack 2 support (I don't believe they have, but havent checked in the last week or so)

**Comment by darkcode85** [comment]
@twalpole Thanks for the quick response, this is the output when I run ```bundle install``` without add ``` gem "sinatra", github: "sinatra/sinatra", branch: "2.2.0-alpha"``` 

```
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      actioncable (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack (~> 2.x) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      actioncable (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack-test (~> 0.6.3) ruby depends on
            rack (>= 1.0) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      actioncable (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack-test (~> 0.6.3) ruby depends on
            rack (>= 1.0) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      sprockets-rails (>= 2.0.0) ruby depends on
        sprockets (>= 3.0.0) ruby depends on
          rack (< 3, > 1) ruby

    omniauth-facebook (>= 0) ruby depends on
      omniauth-oauth2 (~> 1.0.2) ruby depends on
        omniauth (~> 1.0) ruby depends on
          rack (< 3, >= 1.0) ruby

    refile-s3 (>= 0) ruby depends on
      refile (~> 0.5.0) ruby depends on
        sinatra (~> 1.4.5) ruby depends on
          rack (~> 1.4) ruby
```

If I add the sinatra gem in my Gemfile using 2.2.0-alpha version then this is the output: 

```
Bundler could not find compatible versions for gem "rack":
  In Gemfile:
    sinatra (>= 0) ruby depends on
      rack (~> 2.0.0.alpha) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      actioncable (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack (~> 2.x) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      actioncable (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack-test (~> 0.6.3) ruby depends on
            rack (>= 1.0) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      actioncable (= 5.0.0.beta1) ruby depends on
        actionpack (= 5.0.0.beta1) ruby depends on
          rack-test (~> 0.6.3) ruby depends on
            rack (>= 1.0) ruby

    sinatra (>= 0) ruby depends on
      rack-protection (~> 1.5) ruby depends on
        rack (>= 0) ruby

    rails (< 5.1, >= 5.0.0.beta1) ruby depends on
      sprockets-rails (>= 2.0.0) ruby depends on
        sprockets (>= 3.0.0) ruby depends on
          rack (< 3, > 1) ruby

    omniauth-facebook (>= 0) ruby depends on
      omniauth-oauth2 (~> 1.0.2) ruby depends on
        oauth2 (~> 0.8.0) ruby depends on
          rack (~> 1.2) ruby
Bundler could not find compatible versions for gem "sinatra":
  In Gemfile:
    refile-s3 (>= 0) ruby depends on
      refile (~> 0.5.0) ruby depends on
        sinatra (~> 1.4.5) ruby

    sinatra (>= 0) ruby
```

**Comment by twalpole** [comment]
@darkcode85 Since it's showing that in your bundle refile depends on sinatra ~>1.4.5 I have to assume you're not using the master branch of refile since it's gemspec specifies >= 1.4.5

**Pull request merged**